### PR TITLE
adding support of "amazon.titan-embed-text-v2:0"

### PR DESCRIPTION
**Pull request by jdixosnd**
Adding support of newly launched amazon.titan-embed-text-v2:0.
Implemented and tested, working fine.

### CODE REVIEW

File: integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 logger = logging.getLogger(__name__)
 
-SUPPORTED_EMBEDDING_MODELS = ["amazon.titan-embed-text-v1", "cohere.embed-english-v3", "cohere.embed-multilingual-v3"]
+SUPPORTED_EMBEDDING_MODELS = ["amazon.titan-embed-text-v1", "cohere.embed-english-v3", "cohere.embed-multilingual-v3", "amazon.titan-embed-text-v2:0"]
```
Comment (masci) [inline integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py:19]: Can you fix the format of this line, which is too long? You can run `hatch fmt` from the `integrations/amazon_bedrock` folder to see the details, or see the failing CI in this PR https://github.com/deepset-ai/haystack-core-integrations/actions/runs/9206974903/job/25326004022?pr=735

**Review by masci — CHANGES_REQUESTED** [review]

File: integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py (hunk 1 above)
Comment (jdixosnd) [reply integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py:19]: sure let me fix that!

Comment (jdixosnd) [reply integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py]: I've updated the code.

**Review by masci — APPROVED** [review]
Tests passing, thanks for the PR!

**Pull request merged**